code/game/objects/items/storage/boxes.dm

```diff
@@ -605,12 +605,19 @@
 	max_storage_space = 50
 	can_hold = list(/obj/item/explosive/grenade/high_explosive)
 	var/base_icon
+	var/model_icon = "model_m40"
+	var/type_icon = "hedp"
 	var/grenade_type = /obj/item/explosive/grenade/high_explosive
 	has_gamemode_skin = TRUE
 
 /obj/item/storage/box/nade_box/Initialize()
 	. = ..()
 	RegisterSignal(src, COMSIG_ITEM_DROPPED, PROC_REF(try_forced_folding))
+	if(type_icon)
+		overlays += image(icon, type_icon)
+
+	if(model_icon)
+		overlays += image(icon, model_icon)
```

Comment:
Are you sure this behavior needs to change or is doing what you intended?
![image](https://github.com/user-attachments/assets/417a7b90-1562-46bd-9279-e6e69df84044)

Normally they do this: 
![image](https://github.com/user-attachments/assets/a94a8081-8d9f-43eb-b1ce-d010096f3cc4)

Reply:
fixed